hibernate-core/src/main/java/org/hibernate/query/sqm/tree/jpa/CriteriaParameterCollectQueryWalker.java

```diff
@@ -22,24 +25,25 @@
 import org.hibernate.service.ServiceRegistry;
 
 /**
- * todo (6.0) : how is this different from {@link org.hibernate.query.sqm.internal.ParameterCollector}?
+ * A criteria parameter collector to feed parameter collected to a {@link Consumer}
+ * while walking a Semantic Query Model (SQM) tree.
  *
  * @author Steve Ebersole
  */
-public class ParameterCollector extends BaseSemanticQueryWalker {
+public class CriteriaParameterCollectQueryWalker extends BaseSemanticQueryWalker {
```

Comment:
The renaming seems appropriate to avoid the confusion and we can remove a 'todo' item in the Javadoc.